oidc_provider/views.py

```diff
@@ -174,6 +178,20 @@ def post(self, request, *args, **kwargs):
 
             return redirect(uri)
 
+    @staticmethod
+    def strip_prompt_login(path):
+        """
+        Strips 'login' from the 'prompt' query parameter.
+        """
+        uri = urlsplit(path)
+        query_params = parse_qs(uri.query)
+        prompt = query_params['prompt']
```

Comment:
Maybe this part:
```
        prompt = query_params['prompt']
        prompt.remove('login')
        if prompt:
            query_params['prompt'] = prompt
```
change to:

```
        if 'login' in query_params['prompt']:
                query_params['prompt'].remove('login')
```
It is more reliable  (no ValueError exception) and works on references.
In your code, lines `if...` are not necessary and even wrong (when we assume that `prompt` changes reference after `remove` method) because if `login` was only one parameter in `query_params`, this `if` makes that `login` will not be removed from `query_params`. But why it works ? by luck :) `prompt.remove` works on reference so after that line and even `if` logic is wrong, `query_params` are correctly affected.